packages/pwa-kit-react-sdk/src/ssr/universal/contexts/index.js

```diff
@@ -5,9 +5,48 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import React from 'react'
+import React, {useEffect, useRef} from 'react'
+import PropTypes from 'prop-types'
+import {useLocation} from 'react-router-dom'
+import {uuidv4} from '../../../utils/uuidv4.client'
+
 import DeviceContext from '../device-context'
 
 const ExpressContext = React.createContext()
 
-export {DeviceContext, ExpressContext}
+const CorrelationIdContext = React.createContext()
+
+const CorrelationIdProvider = ({children, correlationId, onPageChange = () => {}}) => {
+    const [id, setId] = React.useState(correlationId)
+    const location = useLocation()
+
+    const isFirstRun = useRef(true)
+    const regexp = /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i
```

Comment:
I'm wondering if this is better suited to be a custom prop type validator, or in the future a typescript type. 

Is there a really important reason why we need to validate the prop value? I think it would be safer to just let it be any value so the app doesn't break severely if the header we use from the request object changes its format.